### ISSUE TITLE
Home: Action Bar Content Positioned Before Main Heading 

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#749]
The Home page of the MathShare application has a main ARIA landmark region which correctly wraps all content between the header and footer.  There is also a main heading at level 1, the text of which describes the current problem set being viewed by the user.  However, content is present in the DOM between the opening `<main>` tag and the `<h1>`, which relates to the main page content but is likely to be missed by users navigating by heading. 
 
This content includes sharing-related controls, such as the button to obtain the problem set permalink and links to Google Classroom and Microsoft Teams.  As such, it is directly connected to the specific content located after the heading.

#### Suggested Solution 
 
Ensure that all main page content is placed after the level 1 heading in the DOM.  In this case, the controls to be moved (inside the action bar) are interactive and can receive keyboard focus, so the visual order must also match the placement in the DOM.  That is, the elements cannot be moved in the DOM but then be visually styled to appear before the heading, which would disrupt the tab order of the page and confuse keyboard-only users. 
 
There is already a drop-down menu present on the page when editing or creating a problem set, so it may be a good idea to move these controls into that menu.  Its menu items would then be determined by the current action (edit, review, solve, etc.).

**Comment by johnhbenetech** [comment]
After discussing with @alexrcabral and @benetechMatt 

We should just rearrange the pages to put the h1 above the action bar. So the problem set title should appear first below header (with padding of course).

**Comment by jscholes** [comment]
@rupeshparab - the page order is now correct, but as a result there are no accessible boundaries to separate the action bar from the list of problems.

* The action bar should be preceded by a level 2 heading ("Actions for this Problem Set" or similar).  It can be hidden off-screen (be sr-only) if required, to avoid requiring a design decision.  If there is ever an instance where the action bar is not shown, the heading must not be rendered either.
* There should also be a level 2 heading before the problem list ("Problems in this Set").  Again, sr-only is fine.
* the action bar buttons should be wrapped inside an unordered list, with an ordered list for the problems themselves.
* Each of those lists should have an `aria-labelledby` attribute on their opening tag, the value of which references the ID of the corresponding heading (one of the headings recommended above).  Naturally this implies that the `<h2>` elements both need a unique `id` attribute.

**Comment by johnhbenetech** [comment]
@benetechMatt @alexrcabral 

we'll go with the sr-only approach for the headers, unless we hear from you that you'd like visual text as well

**Comment by rupeshparab** [comment]
I have fixed all of the 4 issues

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/ec2878f896fab8a7bfa3f7c1aa8b4f124ec1ab8d.
